File: java/src/test/java/com/lacombe/promo3/taxi/TaxiOrganizerAcceptanceTest.java
package com.lacombe.promo3.taxi;

import com.lacombe.promo3.meals.CheckIn;
import com.lacombe.promo3.meals.CheckInProvider;
import com.lacombe.promo3.meals.InMemoryCheckInProvider;
import com.lacombe.promo3.meals.RegistrationBook;
import org.junit.Test;

import java.time.LocalDateTime;

import static org.assertj.core.api.Assertions.assertThat;

public class TaxiOrganizerAcceptanceTest {

    @Test
    public void should_have_no_taxi_without_checkin() throws Exception {
        //GIVEN
        final TaxiProvider taxiProvider = new InMemoryTaxiProvider();
        final CheckInProvider inMemoryCheckInProvider = new InMemoryCheckInProvider(RegistrationBook.of());

        final TaxiOrganizer taxiOrganizer = new TaxiOrganizer(taxiProvider, inMemoryCheckInProvider);

        //WHEN
        final TaxiBookingsResult taxiBookingsResult = taxiOrganizer.getBookings();

        //THEN
        assertThat(taxiBookingsResult).isEqualTo(TaxiBookingsResult.notFound());
    }
}

File: dotnet/Socrates.Test/Taxi/TaxiOrganizerAcceptanceTest.cs
using NFluent;
using NUnit.Framework;
using Socrates.Meals;
using Socrates.Taxi;

namespace Socrates.Test.Taxi
{
    public class TaxiOrganizerAcceptanceTest
    {
        [Test]
        public void Should_have_no_taxi_without_checkin()
        {
            //GIVEN
            ITaxiProvider taxiProvider = new InMemoryTaxiProvider();
            ICheckinProvider inMemoryCheckInProvider = new CheckinProvider();

            TaxiOrganizer taxiOrganizer = new TaxiOrganizer(taxiProvider, inMemoryCheckInProvider);

            //WHEN
            TaxiBookingsResult taxiBookingsResult = taxiOrganizer.GetBookings();

            //THEN
            Check.That(taxiBookingsResult).IsEqualTo(TaxiBookingsResult.NotFound());
        }

        [Test]
        public void Should_book_one_regular_taxi_for_period_with_three_checkins()
        {
            //GIVEN
            ITaxiProvider taxiProvider = new InMemoryTaxiProvider();
            ICheckinProvider inMemoryCheckInProvider = new CheckinProvider();

            TaxiOrganizer taxiOrganizer = new TaxiOrganizer(taxiProvider, inMemoryCheckInProvider);

            //WHEN
            TaxiBookingsResult taxiBookingsResult = taxiOrganizer.GetBookings();

            //THEN
            /*Taxi taxi = Taxi.asRegular();
            Collection<TaxiBooking> taxiBookings = Arrays.asList(new TaxiBooking());
            assertThat(taxiBookingsResult).isEqualTo(TaxiBookingsResult.found());*/
        }
    }
}
